src/main/java/org/radarbase/appserver/service/questionnaire/protocol/SimpleProtocolHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *  *
+ *  *  * Copyright 2018 King's College London
+ *  *  *
+ *  *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  *  * you may not use this file except in compliance with the License.
+ *  *  * You may obtain a copy of the License at
+ *  *  *
+ *  *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *  *
+ *  *  * Unless required by applicable law or agreed to in writing, software
+ *  *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *  * See the License for the specific language governing permissions and
+ *  *  * limitations under the License.
+ *  *  *
+ *  *
+ *
+ */
+
+package org.radarbase.appserver.service.questionnaire.protocol;
+
+import org.radarbase.appserver.dto.protocol.Assessment;
+import org.radarbase.appserver.dto.protocol.ReferenceTimestamp;
+import org.radarbase.appserver.dto.questionnaire.AssessmentSchedule;
+import org.radarbase.appserver.entity.User;
+
+import java.time.Instant;
+import java.util.TimeZone;
+
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class SimpleProtocolHandler implements ProtocolHandler {
+    private transient TimeCalculatorService timeCalculatorService = new TimeCalculatorService();
+
+    @Override
+    public AssessmentSchedule handle(AssessmentSchedule assessmentSchedule, Assessment assessment, User user) {
+        ReferenceTimestamp referenceTimestamp = assessment.getProtocol().getReferenceTimestamp();
+        TimeZone timezone = TimeZone.getTimeZone(user.getTimezone());
+        if (referenceTimestamp != null) {
+            switch (referenceTimestamp.getFormat()) {
+                case DATE:
+                case DATETIME:
+                case DATETIMEUTC:
+                    assessmentSchedule.setReferenceTimestamp(Instant.parse(referenceTimestamp.getTimestamp()));
```

Comment:
I think Instant.parse only works on strings containing utc timestamps so something like `2007-12-03T10:15:30.00Z`. That means in our case it will only work for UTC format — containing `Z` at the end.

You should use `LocalDate.parse` for `date` format and `LocalDateTime.parse` for `datetimeutc` (with UTC zoneid) and `datetime` (with user's timezone zoneid) formats.
Something like -
```java
        if (referenceTimestamp != null) {
             switch (referenceTimestamp.getFormat()) {
                 case DATE: 
                    assessmentSchedule.setReferenceTimestamp(LocalDate.parse(referenceTimestamp.getTimestamp()).atStartOfDay(ZoneId.of(timezone)).toInstant());
                    break;
                 case DATETIME:
                    assessmentSchedule.setReferenceTimestamp(LocalDateTime.parse(referenceTimestamp.getTimestamp()).atZone(ZoneId.of(timezone)).toInstant());
                    break;
                 case DATETIMEUTC: 
                  assessmentSchedule.setReferenceTimestamp(LocalDateTime.parse(referenceTimestamp.getTimestamp()).toInstant(ZoneOffset.UTC));
                    break;

```

Reply:
Ah yes my bad I missed that it will only work for UTC format. I'll make the changes.

Reply:
This is working for the following cases:
```
   "referenceTimestamp": {
                "timestamp": "2022-11-09",
                "format": "date"
            },
```
```
      "referenceTimestamp": {
                "timestamp": "2022-11-09T01:00:00.00",
                "format": "datetime"
            },
```
```
  "referenceTimestamp": {
                "timestamp": "2022-11-09T03:00:00.00Z",
                "format": "datetimeutc"
            },
```